enterprise/cmd/frontend/internal/app/app.go

```diff
@@ -146,115 +142,64 @@ func EncryptWithPrivateKey(msg string, privateKey []byte) ([]byte, error) {
+					return
+				}
+
+				var displayName string
+				if ins.Account.Login != nil {
+					displayName = fmt.Sprintf("GitHub (%s)", *ins.Account.Login)
 				}
 
-				if setupAction == "request" {
-					currentConfig, err := svc.Config.Decrypt(context.Background())
-					if err != nil {
-						responseServerError("Failed to edit config", err)
-						return
-					}
-					newConfig, err := jsonc.Edit(currentConfig, true, "pending")
-					if err != nil {
-						responseServerError("Failed to edit config", err)
-						return
-					}
-					svc.Config = extsvc.NewUnencryptedConfig(newConfig)
-				} else if setupAction == "install" {
-					installationID, err := strconv.ParseInt(r.URL.Query().Get("installation_id"), 10, 64)
-					if err != nil {
-						w.WriteHeader(http.StatusBadRequest)
-						_, _ = w.Write([]byte(`The "installation_id" is not a valid integer`))
-						return
-					}
-
-					ins, err := client.GetAppInstallation(r.Context(), installationID)
-					if err != nil {
-						responseServerError(`Failed to get the installation information using the "installation_id"`, err)
-						return
-					}
-
-					if ins.Account.Login != nil {
-						displayName = fmt.Sprintf("GitHub (%s)", *ins.Account.Login)
-					}
-
-					svc.DisplayName = displayName
-					currentConfig, err := svc.Config.Decrypt(context.Background())
-					if err != nil {
-						responseServerError("Failed to edit config", err)
-						return
-					}
-					newConfig, err := jsonc.Edit(currentConfig, strconv.FormatInt(installationID, 10), "githubAppInstallationID")
-					if err != nil {
-						responseServerError("Failed to edit config", err)
-						return
-					}
-					newConfig, err = jsonc.Edit(newConfig, false, "pending")
-					if err != nil {
-						responseServerError("Failed to edit config", err)
-					}
-					svc.Config = extsvc.NewUnencryptedConfig(newConfig)
-					svc.UpdatedAt = now
-					err = db.ExternalServices().Upsert(r.Context(), svc)
-					if err != nil {
-						responseServerError("Failed to upsert code host connection", err)
-						return
-					}
-					encryptedInstallationID, err := EncryptWithPrivateKey(r.URL.Query().Get("installation_id"), privateKey)
-					if err != nil {
-						log15.Error("Error while encrypting installation ID.", "error", err)
-						w.WriteHeader(http.StatusBadRequest)
-						_, _ = w.Write([]byte(`Error while encrypting installation ID`))
-						return
-					}
-					base64InstallationID := base64.StdEncoding.EncodeToString(encryptedInstallationID)
-					http.Redirect(w, r, "/install-github-app-success?installation_id="+url.QueryEscape(base64InstallationID), http.StatusFound)
+				err = createCodeHostConnectionForInstallation(r.Context(), db, installationID, displayName, apiURL.String())
+				if err != nil {
+					responseServerError(http.StatusInternalServerError, "Failed to create code host connection"", err)
```

Comment:
no-no-no, not gonna be fooled by these quotes 😄

```suggestion
					responseServerError(http.StatusInternalServerError, "Failed to create code host connection", err)
```